### PR TITLE
Analysis functions for diatomic molecules. 

### PR DESCRIPTION
**Pull request by Alexsp32**
**Changes:**
- Added new `DynamicsOutputs` for Desorption Angles and DynamicsVariables during desorption (#317)
- Added distance, centre of mass, reduced mass function with and without check for closer periodic copies to a `Structure` submodule. 
- Added an `Analysis` submodule to contain functions needed for more complex analysis of MD trajectories.

### CODE REVIEW

**Review by reinimaurer1 — COMMENTED** [review]
This looks good, although a lot of the utility functions for the output aren't covered by tests. Might be worth to cover the ones that don't to completely drivial things

**Comment by Alexsp32** [comment]
Need to re-work several bits of this PR since they're very memory-inefficient and some edge cases aren't covered.

**Comment by Alexsp32** [comment]
```
DesorptionFrame - No views: 1.191380 seconds (7.33 M allocations: 486.175 MiB, 6.54% gc time, 84.50% compilation time: <1% of which was recompilation)
DesorptionFrame - With views: 0.401479 seconds (4.67 M allocations: 309.999 MiB, 6.78% gc time, 58.73% compilation time)
DesorptionAngle - No views: 0.645341 seconds (5.03 M allocations: 335.521 MiB, 4.93% gc time, 73.09% compilation time: 12% of which was recompilation)
DesorptionAngle - With views: 0.446124 seconds (4.71 M allocations: 313.256 MiB, 4.90% gc time, 61.16% compilation time)
```

I've merged in various changes to decrease the memory footprint of `OutputDesorptionAngle` and `OutputDesorptionTrajectory` and added a unit test for an example system, so I think this is now ready to be merged.

**Comment by reinimaurer1** [comment]
Nicely done!

**Comment by Alexsp32** [comment]
[Docs preview here](https://nqcd.github.io/NQCDynamics.jl/previews/PR329/output_and_analysis/intro/)

File: src/DynamicsOutputs.jl [hunk 1]
```diff
@@ -91,6 +96,21 @@ Evaluate the classical kinetic energy at each timestep during the trajectory.
 OutputKineticEnergy(sol, i) = DynamicsUtils.classical_kinetic_energy.(sol.prob.p, sol.u)
 export OutputKineticEnergy
 
+"""
+    OutputCentroidKineticEnergy(sol, i)
```
Comment (reinimaurer1) [inline src/DynamicsOutputs.jl:100]: please update the docs here

File: src/structure.jl [hunk 2]
```diff
@@ -0,0 +1,183 @@
+module Structure
```
Comment (reinimaurer1) [inline src/structure.jl:1]: can you please create a markdown file for structure.md in docs/API to explain in a few sentences what this module is about

**Review by reinimaurer1 — APPROVED** [review]

**Comment by reinimaurer1** [comment]
@wgst  there are a few new functions that could be useful for you

File: src/structure.jl (hunk 2 above)
Comment (Alexsp32) [reply src/structure.jl:1]: I've slightly improved docs [here](https://github.com/NQCD/NQCDynamics.jl/blob/d7f16e2ea8a07ec4433d8fa66933fa23775886b0/docs/src/api/NQCDynamics/structure.md).
